src/subscription/FeatureListProvider.ts

```diff
@@ -0,0 +1,156 @@
+import Stream from "mithril/stream";
+import {PlanPrices} from "../api/entities/sys/TypeRefs"
+import {TranslationKey} from "../misc/LanguageViewModel"
+
+const FEATURE_LIST_RESOURCE_URL = "https://tutanota.com/resources/data/features.json"
+let dataProvider: HiddenFeatureListProvider | null = null
+
+export interface FeatureListProvider {
+	getFeatureList(targetSubscription: SubscriptionType): FeatureLists[SubscriptionType]
+
+	featureLoadingDone(): boolean
+}
+
+export async function getFeatureListProvider(): Promise<FeatureListProvider> {
+	if (dataProvider == null) {
+		dataProvider = new HiddenFeatureListProvider()
+		await dataProvider.init()
+	}
+	return dataProvider
+}
+
+class HiddenFeatureListProvider implements FeatureListProvider {
```

Comment:
It took me a bit of time to figure out what "Hidden" means here. Why do we do this? Didn't we agree to not make interfaces with one implementation?

Reply:
in this case it's supposed to force the usage of the factory function in the same file to ensure that the implementation is initialized correctly and that we're only getting the features once before proceeding.

PricesAndFeaturesProvider is using a similar pattern. We should probably use private constructors for this.